### DISCUSSION TITLE
how can i add .setThumbnail that shows the video image in nowplaying.js

### DISCUSSION THREAD

**Discussion by kpopdev** [eritislami/evobot#280] [Q&A]
how can i add .setThumbnail that shows the video image in nowplaying.js

**Comment by brandonst** [comment]
You can get the thumbnail of a youtube video with the url
https://img.youtube.com/vi/YourVideoID/maxresdefault.jpg
just replace YourVideoID with that videos id

**Reply by kpopdev** [reply]
how to replace it?i mean if they play random songs

**Reply by brandonst** [reply]
if you get the song from the queue you should be able to get that videos id, then slap the video id into the url

**Comment by Takiyo0** [comment]
in your include/play.js, make an embed first, it will looks like this
```    var playembed = new Discord.MessageEmbed()
    .setTitle("Now Playing")
    .setDescription(`\`${song.title}\``)
    .setThumbnail(song.thumbnail || "https://cdn.iconscout.com/icon/free/png-256/youtube-85-226402.png")
    .addField(`Song's url`, `${song.url}`)
    .setColor('#d539ed')

    try {
      var playingMessage = await queue.textChannel.send(playembed);
      await playingMessage.react("⏭");
      await playingMessage.react("⏹");
    } catch (error) {
      console.error(error);
    }
```
At the right side of song.thumbnail, i add youtube logo link so in case when no thumbnail link is available, it will replace the thumbnail. 
Then, your command/play.js
```if (urlValid) {
      try {
        songInfo = await ytdl.getInfo(url);
        song = {
          title: songInfo.videoDetails.title,
          url: songInfo.videoDetails.video_url,
          duration: songInfo.videoDetails.lengthSeconds,
          thumbnail: songInfo.videoDetails.thumbnail.thumbnails[3].url,
        };
      } catch (error) {
        console.error(error);
        return message.reply(error.message).catch(console.error);
      }
    } else if (scRegex.test(url)) {
      try {
        const trackInfo = await scdl.getInfo(url, SOUNDCLOUD_CLIENT_ID);
        song = {
          title: trackInfo.title,
          url: trackInfo.permalink_url,
          duration: Math.ceil(trackInfo.duration / 1000)
        };
      } catch (error) {
        console.error(error);
        return message.reply(error.message).catch(console.error);
      }
    } else {
      try {
        const results = await youtube.searchVideos(search, 1);
        songInfo = await ytdl.getInfo(results[0].url);
        song = {
          title: songInfo.videoDetails.title,
          url: songInfo.videoDetails.video_url,
          duration: songInfo.videoDetails.lengthSeconds,
          thumbnail: songInfo.videoDetails.thumbnail.thumbnails[3].url,
        };
      } catch (error) {
        console.error(error);
        return message.reply(error.message).catch(console.error);
      }
    }
```
and the last one, in your command/playlist.js
```    const newSongs = videos.map((video) => {
      return (song = {
        title: video.title,
        url: video.url,
        duration: video.durationSeconds,
        thumbnail: video.thumbnails.medium.url
      });
    });
```

it will look like this
![Screen Shot 2020-11-14 at 19 11 52](https://user-images.githubusercontent.com/60313514/99146737-5f991e00-26ad-11eb-8320-d85110012b4c.png)


Correct me if i wrong :)

**Reply by Minecraft-Boy** [reply]
> in your include/play.js, make an embed first, it will looks like this
> 
> ```
>     .setTitle("Now Playing")
>     .setDescription(`\`${song.title}\``)
>     .setThumbnail(song.thumbnail || "https://cdn.iconscout.com/icon/free/png-256/youtube-85-226402.png")
>     .addField(`Song's url`, `${song.url}`)
>     .setColor('#d539ed')
> 
>     try {
>       var playingMessage = await queue.textChannel.send(playembed);
>       await playingMessage.react("⏭");
>       await playingMessage.react("⏹");
>     } catch (error) {
>       console.error(error);
>     }
> ```
> 
> At the right side of song.thumbnail, i add youtube logo link so in case when no thumbnail link is available, it will replace the thumbnail.
> Then, your command/play.js
> 
> ```
>       try {
>         songInfo = await ytdl.getInfo(url);
>         song = {
>           title: songInfo.videoDetails.title,
>           url: songInfo.videoDetails.video_url,
>           duration: songInfo.videoDetails.lengthSeconds,
>           thumbnail: songInfo.videoDetails.thumbnail.thumbnails[3].url,
>         };
>       } catch (error) {
>         console.error(error);
>         return message.reply(error.message).catch(console.error);
>       }
>     } else if (scRegex.test(url)) {
>       try {
>         const trackInfo = await scdl.getInfo(url, SOUNDCLOUD_CLIENT_ID);
>         song = {
>           title: trackInfo.title,
>           url: trackInfo.permalink_url,
>           duration: Math.ceil(trackInfo.duration / 1000)
>         };
>       } catch (error) {
>         console.error(error);
>         return message.reply(error.message).catch(console.error);
>       }
>     } else {
>       try {
>         const results = await youtube.searchVideos(search, 1);
>         songInfo = await ytdl.getInfo(results[0].url);
>         song = {
>           title: songInfo.videoDetails.title,
>           url: songInfo.videoDetails.video_url,
>           duration: songInfo.videoDetails.lengthSeconds,
>           thumbnail: songInfo.videoDetails.thumbnail.thumbnails[3].url,
>         };
>       } catch (error) {
>         console.error(error);
>         return message.reply(error.message).catch(console.error);
>       }
>     }
> ```
> 
> and the last one, in your command/playlist.js
> 
> ```
>       return (song = {
>         title: video.title,
>         url: video.url,
>         duration: video.durationSeconds,
>         thumbnail: video.thumbnails.medium.url
>       });
>     });
> ```
> 
> it will look like this
> ![Screen Shot 2020-11-14 at 19 11 52](https://user-images.githubusercontent.com/60313514/99146737-5f991e00-26ad-11eb-8320-d85110012b4c.png)
> 
> Correct me if i wrong :)

For some reason, it doesn't. Do u think you can copy-paste all the code form the files from top to bottom so I don't have to think y its not working pls.
